tests/functional/test_upgrade_actions.py

```diff
@@ -215,3 +217,45 @@ def _extract_configs(cluster: Cluster):
             for service in cluster.service_list()
         },
     }
+
+
+class TestUpgradeActionRelations:
+    """Test cases when upgrade action"""
+
+    @pytest.mark.parametrize(
+        ("folder_dir", "file_dir"),
+        [
+            ('upgrade_failures', 'before_switch'),
+            ('upgrade_success', 'after_switch'),
+        ],
+    )
+    def test_check_upgrade_actions_relations(self, sdk_client_fs, folder_dir, file_dir):
+        """
+        Test bundle action fails before bundle_switch was performed
+        """
+        bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, folder_dir, 'old'))
+        cluster = bundle.cluster_create('Test Cluster for Upgrade')
+        sdk_client_fs.upload_from_fs(get_data_dir(__file__, folder_dir, file_dir))
+        with allure.step("Check cluster actions list before update"):
+            actions_before = cluster.action_list().data
+            assert len(actions_before) == 1, "Should be 1 action"
+            assert actions_before[0].display_name == "dummy_action", "Should be action 'dummy_action'"
+        cluster.upgrade().do().wait()
+        cluster.reread()
+        with allure.step("Check jobs"):
+            jobs = sdk_client_fs.job_list()
+            assert len(jobs) >= 3, "There are should be 3 or more jobs"
```

Comment:
same with logfiles

Reply:
no, because 3 only with clean adcm

Reply:
that's why I suggest running this test only on clean adcm or use value of 3 + jobs already created.
because if on dirty ADCM jobs will fail to be created, we won't know it, because we have non-strict check

Reply:
changed

Reply:
changed

Reply:
see comment below

Reply:
changed